netbox/utilities/api.py

```diff
@@ -87,6 +88,14 @@ def get_prefetches_for_serializer(serializer_class, fields_to_include=None):
         fields_to_include = serializer_class.Meta.fields
 
     prefetch_fields = []
+
+    # If this serializer is nested, get annotations and prefetches for the nested serializer
+    if source_field is not None:
+        nested_annotations = get_annotations_for_serializer(serializer_class, fields_to_include=fields_to_include)
+        if nested_annotations:
+            related_prefetch = Prefetch(source_field, queryset=model.objects.all().annotate(**nested_annotations))
+            prefetch_fields.append(related_prefetch)
```

Comment:
Almost there - I think you can just add a return prefetch_fields here.  If it's gotten to here then in the loop below on line 99 the only time it modifies prefetch_fields is on line 125 and that can only happen if source_field is None, which it wouldn't be in this case, so all that code is looping / processing for nothing.

If that is the case the fields_to_include doesn't matter in this case.  I'm seeing when it is called with source_field you are only getting one value returned for prefetch_fieds (which makes sense as the for loop doesn't append anything) Therefore it doesn't look like you need to recurse into this at all, the lines above can just be a separate function that is called (without the fields_to_include). 

Please let me know if that isn't clear.